### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by jschueller**
Mainly optimization fixes this time:
- Fix OptimizationResult always accepting first point regardless of constraint
- Disable NLopt PRAXIS/AGS/STOGO (nlopt bugs) & run with deterministic seed (SetSeed => setSeed)
- Fix for DLib maximization with version<19.19
- enchanced MultiStart to take constraint violation under account

Also
- Allow stop redirecting Log to file
- #2354 : add a Comparison operator to the class instead hardwiring (X>T)
- Drop matrix/tensor implementation classes from SWIG
- Fix for Sample parsing spaces in csv labels

### CODE REVIEW

**Review by sofianehaddad — APPROVED** [review]

**Comment by jschueller** [comment]
@regislebrun did you have a look at this one ? I'd really like to merge it before starting another branch

File: lib/src/Base/Optim/Dlib.cxx [hunk 1]
```diff
@@ -645,28 +640,30 @@ void Dlib::run()
...
+//                                         ub);
+//       else
+//         dlib::find_max (searchStrategy,
+//                         stopStrategy,
+//                         objectiveDlibFunction,
+//                         objectiveDlibGradient,
+//                         optimPoint,
+//                         SpecFunc::MaxScalar);
+//     }
+
```
Comment (regislebrun) [inline lib/src/Base/Optim/Dlib.cxx:661]: Dead code?

File: lib/src/Base/Optim/openturns/DlibFunctions.hxx [hunk 2]
```diff
@@ -43,15 +43,16 @@ class DlibGradient
 {
 public:
   // Constructor
-  explicit DlibGradient(const Gradient & gradient);
+  DlibGradient(const Gradient & gradient, const Bool minimization = true);
```
Comment (regislebrun) [inline lib/src/Base/Optim/openturns/DlibFunctions.hxx:46]: Why did you removed the explicit specifier?

File: lib/src/Base/Optim/openturns/DlibFunctions.hxx [hunk 3]
```diff
@@ -60,10 +61,11 @@ class DlibHessian
 {
 public:
   // Constructor
-  explicit DlibHessian(const Hessian & hessian);
+  DlibHessian(const Hessian & hessian);
```
Comment (regislebrun) [inline lib/src/Base/Optim/openturns/DlibFunctions.hxx:64]: same here

**Review by regislebrun — APPROVED** [review]
Only one question for curiosity ;-)

File: lib/src/Base/Optim/openturns/DlibFunctions.hxx (hunk 2 above)
Comment (jschueller) [reply lib/src/Base/Optim/openturns/DlibFunctions.hxx:46]: seems like a mistake

File: lib/src/Base/Optim/Dlib.cxx (hunk 1 above)
Comment (jschueller) [reply lib/src/Base/Optim/Dlib.cxx:620]: yes, replace that with an explanation of why we dont use find_max

**Pull request merged**